go/vt/wrangler/tablet.go

```diff
@@ -70,9 +72,12 @@ func (wr *Wrangler) InitTablet(ctx context.Context, tablet *topodatapb.Tablet, a
 		return fmt.Errorf("creating this tablet would override old master %v in shard %v/%v, use allow_master_override flag", topoproto.TabletAliasString(si.MasterAlias), tablet.Keyspace, tablet.Shard)
 	}
 
-	// update the shard record if needed
-	if err := wr.updateShardMaster(ctx, si, tablet.Alias, tablet.Type, allowMasterOverride); err != nil {
-		return err
+	if tablet.Type == topodatapb.TabletType_MASTER {
+		if !si.HasMaster() /* no master */ || (si.HasMaster() && topoproto.TabletAliasEqual(si.MasterAlias, tablet.Alias)) /* same master*/ || (si.HasMaster() && !topoproto.TabletAliasEqual(si.MasterAlias, tablet.Alias) && allowMasterOverride) /* override current master */ {
+			// we update master_term_start_time even if the master hasn't changed
+			// because that means a new master term with the same master
+			tablet.MasterTermStartTime = logutil.TimeToProto(time.Now())
+		}
```

Comment:
@enisoc do you think this covers all possibilities?

Reply:
There's already a bailout above after checking for an existing master. If we make it past that, can't we assume it's always safe to set `MasterTermStartTime` whenever the tablet type is MASTER?